src/downloader/api_response_parser.py

```diff
@@ -0,0 +1,97 @@
+import logging
+import mypy.api
+
+from abc import ABCMeta, abstractmethod
+from collections import defaultdict
+from datetime import datetime
+
+from .batch_objects import BatchResponse
+from .config import VACANCY_TYPES
+
+
+logging.basicConfig(format='%(asctime)s %(levelname)s %(message)s', level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+
+class APIResponseParser(metaclass=ABCMeta):
+
+    def __init__(self, types: dict = VACANCY_TYPES) -> None:
+        self.types = types
+        self.fields = tuple(types.keys())
+
+    def map_fields(self, sample: dict) -> dict:
+        default_dict = defaultdict(lambda: None, sample)
+        result = {}
+        for key in self.fields:
```

Comment:
Вот здесь например можно итерироваться просто по ключам словаря
```suggestion
        for key in self.types:
```